packages/server/validators/events.js

```diff
@@ -203,7 +203,18 @@ const attendeeStatusValidator = [
     .matches(/^(NotInterested|Interested|Attending)$/)
     .withMessage(
       "Invalid status value. Allowed values are: NotInterested, Interested, Attending"
-    ),
+    )
+    .custom(async (eventId, { req }) => {
+      const userId = req.body.userId;
+      const event = await EventController.getEvent(eventId);
+      const isMember = await GuildController.isGuildMember(
+        event.guildId,
+        userId
+      );
+      if (!isMember) {
+        throw new Error("User is not a member of the guild.");
+      }
+    }),
```

Comment:
This is the right idea but the value inside the async function parameter is not `eventId` in this case, it's `status`. The value that's passed to the handler function inside a custom validator is the field that the custom validator is attached to, `body("status")` in this case.

`eventId` can be accessed by `req.params.eventId` inside the async function since the `eventIdValidator` is ran before this in the route handler function in `routes/api/events.js` so `eventId is already validated.

```js
.custom(async (status, { req }) => {
    const userId = req.body.userId;
    const eventId = req.params.eventId;    // access eventId here
    const event = await EventController.getEvent(eventId);
    const isMember = await GuildController.isGuildMember(
      event.guildId,
      userId
    );
    if (!isMember) {
      throw new Error("User is not a member of the guild.");
    }
  })
```

See docs [here](https://express-validator.github.io/docs/guides/customizing#implementing-a-custom-validator) for custom validators.